agent/engine/docker_image_manager_test.go

```diff
@@ -71,7 +71,9 @@ func TestImagePullRemoveDeadlock(t *testing.T) {
 	}()
 
 	go func() {
-		imageManager.(*dockerImageManager).removeUnusedImages()
+		ctx, cancel := context.WithCancel(context.TODO())
```

Comment:
can you move the context creation and `defer cancel()` methods outside of the go routine?

Reply:
OK